apps/base-docs/docs/pages/identity/smart-wallet/guides/linked-accounts/using-sub-accounts.mdx

```diff
@@ -0,0 +1,235 @@
+---
+title: 'Using Sub Accounts'
+description: 'Learn how to perform operations with sub accounts in Coinbase Smart Wallet'
+---
+
+# Using Sub Accounts
+
+Once you've created a sub account, you can use it to perform various operations like signing messages and sending transactions. In this section, we'll cover how to interact with your sub account.
+
+## Setting up the Sub Account Client
+
+First, we need to update our context to include a wallet client specifically for the sub account. Add the following to your CoinbaseWalletContext.tsx:
+
+```tsx
+// app/context/CoinbaseWalletContext.tsx
+import {
+  createCoinbaseWalletSDK,
+  getCryptoKeyAccount,
+  WebAuthnAccount,
+  toCoinbaseSmartAccount,
+} from '@coinbase/wallet-sdk';
+import { WalletClient, createPublicClient, http } from 'viem';
+
+export function CoinbaseWalletProvider({ children }: { children: ReactNode }) {
+  // ... existing state
+  const [subAccountWalletClient, setSubAccountWalletClient] = useState<WalletClient | null>(null);
+
+  // Add public client for account creation
+  const publicClient = useMemo(
+    () =>
+      createPublicClient({
+        chain: baseSepolia,
+        transport: http(),
+      }),
+    [],
+  );
+
+  // Initialize sub account client when sub account is created
+  useEffect(() => {
+    async function initializeSubAccountClient() {
+      if (!subAccount || !provider || !publicClient) {
+        setSubAccountWalletClient(null);
+        return;
+      }
+
+      try {
+        const signer = await getCryptoKeyAccount();
+        if (!signer) {
+          throw new Error('Signer not found');
+        }
+
+        const account = await toCoinbaseSmartAccount({
+          client: publicClient,
+          owners: [signer.account as WebAuthnAccount],
+          address: subAccount,
+        });
```

Comment:
Is this needed? I think we should be okay to just proxy requests via the provider

Reply:
My fear with recommending this is this fails with ownership changes or if the account isn't  deployed